src/dataflow/src/render/debezium.rs

```diff
@@ -118,6 +124,273 @@ where
     }
 }
 
+pub(crate) fn render_tx<G: Scope>(
+    data_id: GlobalId,
+    envelope: &DebeziumEnvelope,
+    input: &Stream<G, DecodeResult>,
+    tx_ok: Collection<G, Row, Diff>,
+    debug_name: String,
+) -> (
+    Stream<G, (Row, Timestamp, Diff)>,
+    Stream<G, (mz_dataflow_types::DataflowError, Timestamp, Diff)>,
+)
+where
+    G: ScopeParent<Timestamp = Timestamp>,
+{
+    let (before_idx, after_idx) = (envelope.before_idx, envelope.after_idx);
+    let hashed_id = data_id.hashed();
+
+    let (transaction_idx, tx_id_idx) = match envelope {
+        DebeziumEnvelope {
+            mode:
+                DebeziumMode::Full(DebeziumDedupProjection {
+                    transaction_idx,
+                    tx_id_idx,
+                    ..
+                }),
+            ..
+        }
+        | DebeziumEnvelope {
+            mode:
+                DebeziumMode::Ordered(DebeziumDedupProjection {
+                    transaction_idx,
+                    tx_id_idx,
+                    ..
+                }),
+            ..
+        }
+        | DebeziumEnvelope {
+            mode:
+                DebeziumMode::FullInRange {
+                    projection:
+                        DebeziumDedupProjection {
+                            transaction_idx,
+                            tx_id_idx,
+                            ..
+                        },
+                    ..
+                },
+            ..
+        } => (
+            *transaction_idx,
+            tx_id_idx.expect("render_tx should only be called when there's a transaction"),
```

Comment:
do we validate this somehow before? 1 bad message from debezium could cause a panic, right?